lib/rom/sql/associations/core.rb

```diff
@@ -14,6 +14,14 @@ def preload(target, loaded)
 
           target.where(target_key => target_pks)
         end
+
+        # @api private
+        def wrapped
+          new_target = view ? target.send(view) : target
+          to_wrap = self.class.allocate
+          to_wrap.send(:initialize, definition, options.merge(target: new_target))
+          to_wrap.wrap
+        end
```

Comment:
Could you remind me why you had to use `Object#allocate` here?

Reply:
`Association#new` is overridden with defaults that do not work for this use case

Reply:
@ianks could we tweak the underlying implementation in a way that would not require using `allocate` + `send(:initialize, ...)` here? something like `new(target: view ? target.send(view) : target)` should work.

Reply:
...sorry, I meant `with(target: view ? target.send(view).target)`, that's the conventional method we use in rom.

Reply:
does this require changes to rom-core? @solnic

Reply:
@ianks OK I see what's going on, it's due to the way `Associations::Core.new` is implemented. This is unfortunate, I'll do something about it later so for now let's leave this `allocate` trick here.